app/models/patient.rb

```diff
@@ -146,17 +146,21 @@ def latest_transfer
 
   # Most recent vaccine dosage
   def latest_dosage
-    dosages.order(created_at: :desc).first
+    # dosages.order(created_at: :desc).first
+    dosages.order(date_given: :desc).first
   end
 
   # Patients who are eligible for reminders
+  # TODO: Right now we're going to just assume that you just need a dosage. So we'll do a simple join
   scope :reminder_eligible, lambda {
     where(purged: false)
+      .joins(:dosages)
       .where(pause_notifications: false)
       .where('patients.id = patients.responder_id')
       .where.not('latest_assessment_at >= ?', Time.now.in_time_zone('Eastern Time (US & Canada)').beginning_of_day)
       .or(
         where(purged: false)
+          .joins(:dosages)
           .where(pause_notifications: false)
           .where('patients.id = patients.responder_id')
           .where(latest_assessment_at: nil)
```

Comment:
Can we change this to the below? It seems like the whole OR block is just to check if the latest assessment is null when we could just check both in the same where check.
```
scope :reminder_eligible, lambda {
    where(purged: false)
      .joins(:dosages)
      .where(pause_notifications: false)
      .where('patients.id = patients.responder_id')
      .where('last_assessment_reminder_sent IS NULL OR latest_assessment_at <= ?', Time.now.in_time_zone('Eastern Time (US & Canada)').beginning_of_day)
      .distinct
  }
```

Reply:
Changed in 93ea627